src/modules/users/users.controller.ts

```diff
@@ -1,32 +1,138 @@
 import { UserService } from './users.service';
-import { Controller, Get, Post, Put, Delete } from '@nestjs/common';
+import {
+  Controller,
+  Get,
+  Put,
+  Delete,
+  UseGuards,
+  Body,
+  UseInterceptors,
+  UploadedFile,
+  Param,
+  NotFoundException,
+} from '@nestjs/common';
+import { UpdateProfileDto } from './dto/requests/update-profile.dto';
+import { RolesGuard, UserRoles } from '#/shared/guards/roles.guard';
+import { Roles } from '#/shared/constants/user-roles.constant';
+import { withBaseResponse } from '#/shared/utils/with-base-response.util';
 
-@Controller('user')
+@Controller('users')
 export class UserController {
-  constructor(private readonly userService: UserService) {}
+  constructor(
+    private readonly userService: UserService,
+    private readonly uploadService: UploadService,
+  ) {}
 
-  @Post()
-  create() {
-    return this.userService.create();
+  @Get()
+  @UseGuards(RolesGuard)
+  @UserRoles(Roles.Admin)
+  @UseGuards(AccessTokenGuard)
+  async listUsers() {
+    const result = await this.userService.listUsers();
+
+    return withBaseResponse({
+      status: 200,
+      message: 'Users successfully received',
+      data: result,
+    });
   }
 
-  @Get()
-  findAll() {
-    return this.userService.findAll();
+  @Get('profile')
+  @UseGuards(AccessTokenGuard)
+  async getProfile(@GetAccount('accountId') userId: string) {
+    const result = await this.userService.getProfile(userId);
+
+    return withBaseResponse({
+      status: 200,
+      message: 'Profile successfully received',
+      data: result,
+    });
   }
 
-  @Get()
-  findOne() {
-    return this.userService.findOne();
+  @Get('wishlist')
+  @UseGuards(AccessTokenGuard)
+  async getUserWishlist(@GetAccount('accountId') userId: string) {
+    const result = await this.userService.getUserWishlist(userId);
+
+    return withBaseResponse({
+      status: 200,
+      message: 'User wishlist successfully received',
+      data: result,
+    });
   }
 
-  @Put()
-  update() {
-    return this.userService.update();
+  @Put('profile')
+  @UseGuards(AccessTokenGuard)
+  async updateProfile(
+    @GetAccount('accountId') userId: string,
+    @Body() updateProfileDto: UpdateProfileDto,
+  ) {
+    const result = await this.userService.updateProfile(
+      userId,
+      updateProfileDto,
+    );
+
+    return withBaseResponse({
+      status: 200,
+      message: 'Profile successfully updated',
+      data: result,
+    });
   }
 
-  @Delete()
-  remove() {
-    return this.userService.remove();
+  @Put('profile/avatar')
+  @UseGuards(AccessTokenGuard)
+  @UseInterceptors(FileInterceptor('image'))
+  async updateProfileAvatar(
+    @UploadedFile() image: Express.Multer.File,
+    @GetAccount('accountId') userId: string,
+  ) {
+    if (!image) {
+      throw new NotFoundException('Uploading file not found');
+    }
+
+    const imgUrl = await this.uploadService.uploadImage(image);
```

Comment:
This would become ```avatarUrl```. In the future, if we want to add banner photo as well for a profile, we could easily distinguish what is what, instead of calling them all ```image```

Reply:
Fixed